storage/block/block.go

```diff
@@ -391,16 +390,7 @@ func (b *dbBlock) Close() {
 	}
 
 	b.closed = true
-
-	// NB(xichen): we use the worker pool to close the context instead doing
-	// an asynchronous context close to explicitly control the context closing
-	// concurrency. This is particularly important during a node removal because
-	// all the shards are removed at once, causing a goroutine explosion without
-	// limiting the concurrency. We also cannot do a blocking close here because
-	// the block may be closed before the underlying context is closed, which
-	// causes a deadlock if the block and the underlying context are closed
-	// from within the same goroutine.
-	b.opts.CloseContextWorkers().Go(b.ctx.BlockingClose)
```

Comment:
Aren't we just leaking the ctx by removing this?

Reply:
I think the removed the context all together.

Reply:
Yup context is gone, we now just copy the bytes each time rather than depending on the caller's context and taking ref.